pointercrate-user/src/auth/mod.rs

```diff
@@ -208,7 +213,11 @@ impl AuthenticatedUser {
     }
 
     fn password_salt(&self) -> Vec<u8> {
-        let raw_parts: Vec<_> = self.password_hash.split('$').filter(|s| !s.is_empty()).collect();
+        if self.password_hash.is_none() {
+            return Vec::new();
+        }
+
+        let raw_parts: Vec<_> = self.password_hash.as_ref().unwrap().split('$').filter(|s| !s.is_empty()).collect();
```

Comment:
let's use `match` instead of `unwrap` here:
```suggestion
        let raw_parts = match self.password_hash {
            None => return Vec::new(),
            Some(ref hash) => hash.split('$').filter(|s| !s.is_empty()).collect();
        };
```